src/timedelta_isoformat/__init__.py

```diff
@@ -1,19 +1,35 @@
 """Supplemental ISO8601 duration format support for :py:class:`datetime.timedelta`"""
 import datetime
-from typing import Iterable, Tuple, TypeAlias
+from typing import Iterable, Tuple, TypeAlias, Dict
+from dataclasses import dataclass
 
 _DIGITS, _DECIMAL_SIGNS = frozenset("0123456789"), frozenset(",.")
 _FORMAT = _DIGITS | _DECIMAL_SIGNS
 
 
+Token: TypeAlias = str
+TimedeltaArgument: TypeAlias = str
+UnparsedValue: TypeAlias = str
+ValueLimit: TypeAlias = int | None
+Components: TypeAlias = Iterable[Tuple[UnparsedValue, TimedeltaArgument, ValueLimit]]
+Measurements: TypeAlias = Iterable[Tuple[TimedeltaArgument, float]]
+
+class ParsingContext:
+    remaining_tokens: Iterable[Token]
+    token_to_timedelta_arg: Dict[Token, TimedeltaArgument]
```

Comment:
The reasoning behind this idea is: we have a class that represents a one-to-one mapping from a named value (`Y` / `M` / `D` in the case of the date context) to corresponding units (`years`, `months`, `days`) - and it's a static mapping.  So it could be a good candidate for an enumeration type.

Also: fun fact: `enum` types support iteration over the keys, and if I remember correctly, that iteration preserves key order.